cli/cmd/publicdomain/add.go

```diff
@@ -0,0 +1,66 @@
+package publicdomain
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/norman/pkg/kv"
+	"github.com/rancher/rio/cli/pkg/clicontext"
+	"github.com/rancher/rio/types/client/space/v1beta1"
+)
+
+type Add struct {
+}
+
+func (a *Add) Run(ctx *clicontext.CLIContext) error {
+	if len(ctx.CLI.Args()) != 3 {
+		return errors.New("Incorrect Usage. Example: `rio domain add NAME DOMAIN_NAME TARGET_SVC`")
+	}
+	name := ctx.CLI.Args().Get(0)
+	domainName := ctx.CLI.Args().Get(1)
+	target := ctx.CLI.Args().Get(2)
+	cluster, err := ctx.Cluster()
+	if err != nil {
+		return err
+	}
+	spaceClient, err := cluster.Client()
+	if err != nil {
+		return err
+	}
+	space, stack, targetName := resolveSpaceStackName(target, cluster.DefaultWorkspaceName)
+	domain := &client.PublicDomain{
+		Name:                name,
+		DomainName:          domainName,
+		TargetWorkspaceName: space,
+		TargetStackName:     stack,
+		TargetName:          targetName,
+	}
+	domain, err = spaceClient.PublicDomain.Create(domain)
+	if err != nil {
+		return err
+	}
+	fmt.Println(domain.Name)
+	return nil
+}
+
+func resolveSpaceStackName(name, defaultWorkspace string) (string, string, string) {
```

Comment:
Is this the same as rio/cli/pkg/stack/ResolveSpaceStackForName?

Reply:
not exactly same. I copied the method and changed it so that it can resolve to workspace/stack/name.